src/binaryManager.ts

```diff
@@ -93,33 +100,76 @@ async function getConfiguredMirrordBinary(): Promise<string | null> {
+
     if (configured) {
+        vscode.window.showInformationMessage(`Using mirrord binary specified in settings: ${configured}`);
         return configured;
     }
+    const latestVersion = await getLatestSupportedVersion(10000);
+    const autoUpdateConfigured = vscode.workspace.getConfiguration().get("mirrord.autoUpdate");
+    const extensionPath = getExtensionMirrordPath().fsPath;
 
-    let foundLocal = await getLocalMirrordBinary();
-    // timeout is 1s if we have alternative or 10s if we don't
-    let timeout = foundLocal ? 1000 : 10000;
-    const latestVersion = await getLatestSupportedVersion(timeout);
-
-    // See if maybe we have it installed already, in correct version.
-    const localMirrord = await getLocalMirrordBinary(latestVersion);
-    if (localMirrord) {
-        const api = new MirrordAPI(localMirrord);
-        const installedVersion = await api.getBinaryVersion();
-        if (installedVersion === latestVersion) {
-            return localMirrord;
+    // if extension is activating, then set a global state to what was read in the workspace settings
+    if (extensionActivate) {
+        globalContext.workspaceState.update('autoUpdate', autoUpdateConfigured);
+    } else {
+        // extension is active, check if auto-update setting has changed
+        const autoUpdate = globalContext.workspaceState.get('autoUpdate');
+        if (autoUpdate !== autoUpdateConfigured) {
+            await new NotificationBuilder()
+                .withMessage("mirrord binary auto-update setting has changed, please reload the window for the change to take effect.")
+                .withGenericAction("Reload", async () => {
+                    await vscode.commands.executeCommand("workbench.action.reloadWindow");
+                })
+                .warning();
         }
+        return extensionPath;
     }
 
-    const extensionMirrordPath = getExtensionMirrordPath();
-    await downloadMirrordBinary(extensionMirrordPath, latestVersion);
+    if (typeof autoUpdateConfigured === 'string') {
+        if (semver.valid(autoUpdateConfigured)) {
+            const localMirrordBinary = await getLocalMirrordBinary(autoUpdateConfigured);
+            if (localMirrordBinary) {
+                return localMirrordBinary;
+            }
+            await downloadMirrordBinary(getExtensionMirrordPath(), autoUpdateConfigured);
+            return extensionPath;
+        } else {
+            vscode.window.showErrorMessage(`Invalid version format ${autoUpdateConfigured}: must follow semver format`);
+        }
+    }
 
-    return extensionMirrordPath.fsPath;
+    if (typeof autoUpdateConfigured === 'boolean') {
+        if (!autoUpdateConfigured) {
+            const localMirrordBinary = await getLocalMirrordBinary();
+            if (localMirrordBinary) {
+                return localMirrordBinary;
+            }
+            return extensionPath;
+        }
+    }
+
+    // Auto-update is enabled or no specific version specified.    
```

Comment:
wdym "no specific version specified"? In what condition will this code be reached if auto update isn't enabled?

Reply:
old comment, please ignore